Mathlib/Algebra/MonoidAlgebra/Defs.lean

```diff
@@ -574,6 +574,31 @@ theorem liftNC_smul [MulOneClass G] {R : Type*} [Semiring R] (f : k →+* R) (g
 
 end MiscTheorems
 
+section Linear
+
+variable {A : Type*} [Semiring k] [Semiring A] [Module k A]
+
+variable (k A) in
+/-- Interpret `single i` as a linear map (but with `MonoidAlgebra` in the type, rather than
+`Finsupp`). -/
+def lsingle (i : G) : A →ₗ[k] MonoidAlgebra A G :=
+  Finsupp.lsingle i
+
+lemma lsingle_def (i : G) :
+    lsingle k A i = Finsupp.lsingle i := rfl
+
+@[simp]
+lemma lsingle_apply (i : G) (a : A) :
+    lsingle k A i a = single i a := rfl
+
+@[ext]
+theorem lhom_ext {B : Type*} [AddCommMonoid B] [Module k B]
+    {f g : MonoidAlgebra A G →ₗ[k] B}
+    (h : ∀ a b, f (lsingle k A a b) = g (lsingle k A a b)) : f = g :=
```

Comment:
This should be stated with function composition, per "partially-applied ext lemmas". I  believe the proof is then `Finsupp.lhom_ext'`